### PR TITLE
Http exception refactoring

### PR DESCRIPTION
**Pull request by tomas-langer**
HttpException moved to common http module.
RequestException (new thing) moved to common http module.

Document describing how exceptions should be handled in HTTP added to repository (internal docs).

`ErrorHandlers` added to Nima (to be fully implemented later - already on task list).

### CODE REVIEW

File: nima/webserver/webserver/src/main/java/io/helidon/nima/webserver/http/ErrorHandlers.java [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+            if (hasErrorHandler(e.getCause())) {
+                handleError(ctx, request, response, e.getCause());
+            } else {
+                handleError(ctx, request, response, e);
+            }
+        } catch (HttpException e) {
+            handleError(ctx, request, response, e);
+        } catch (RuntimeException e) {
+            handleError(ctx, request, response, e);
+        }
```
Comment (danielkec) [inline nima/webserver/webserver/src/main/java/io/helidon/nima/webserver/http/ErrorHandlers.java:74]: I would rather catch Throwable to be sure

**Review by danielkec — DISMISSED** [review]

File: common/http/src/main/java/io/helidon/common/http/ForbiddenException.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.common.http;
+
+/**
+ * A runtime exception indicating a {@link io.helidon.common.http.Http.Status#NOT_FOUND_404 not found}.
```
Comment (ljnelson) [inline common/http/src/main/java/io/helidon/common/http/ForbiddenException.java:20]: Did you mean 403?

File: nima/webserver/webserver/src/main/java/io/helidon/nima/webserver/http/ErrorHandlers.java (hunk 1 above)
Comment (tomas-langer) [reply nima/webserver/webserver/src/main/java/io/helidon/nima/webserver/http/ErrorHandlers.java:74]: I don't think we want to handle "Throwable" - that would contain things like `OutOfMemoryError` and that should not be a case for exception handling by users. We can add it once we add error handling to Nima...

File: common/http/src/main/java/io/helidon/common/http/ForbiddenException.java (hunk 2 above)
Comment (tomas-langer) [reply common/http/src/main/java/io/helidon/common/http/ForbiddenException.java]: Fixed

**Review by barchetta — APPROVED** [review]

**Pull request merged**